### PR TITLE
Remove Servlet dependency on websockets

### PR DESCRIPTION
**Pull request by stuartwdouglas**
This will allow websockets to be used without
Undertow Servlet.

### CODE REVIEW

**Comment by stuartwdouglas** [comment]
@cescoffier is this going to cause problems for the Vert.x 4 work? It will require a 3.1 release as it changes the module structure.

**Comment by cescoffier** [comment]
It will need to be backported to the vertx 4 branch, and another beta would need to be cut. I can handle that today.

File: websocket/core/src/main/java/io/undertow/websockets/handshake/Base64.java [hunk 1]
```diff
@@ -12,15 +12,15 @@
...
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 
-import io.undertow.UndertowLogger;
-
 /**
  * <p>
  * Encodes and decodes to and from Base64 notation.
```
Comment (cescoffier) [inline websocket/core/src/main/java/io/undertow/websockets/handshake/Base64.java:26]: Any reason to not use the Base64 encoders/decoders provided by the JDK?

File: websocket/vertx/src/main/java/io/undertow/websockets/vertx/VertxWebSocketHandler.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                    }
+                }
+
+                if (handshaker != null) {
+                    if (container.isClosed()) {
+                        resp.setStatusCode(HttpResponseStatus.SERVICE_UNAVAILABLE.code()).end();
+                        return;
+                    }
+                    facade.putAttachment(HandshakeUtil.PATH_PARAMS, matchResult.getParameters());
+                    //facade.putAttachment(HandshakeUtil.PRINCIPAL, req.getUserPrincipal());
```
Comment (cescoffier) [inline websocket/vertx/src/main/java/io/undertow/websockets/vertx/VertxWebSocketHandler.java:103]: To be removed?

**Comment by cescoffier** [comment]
It looks good to me. The work to get it working with Vert.x 4 is not big (hopefully).

File: websocket/core/src/main/java/io/undertow/websockets/handshake/Base64.java (hunk 1 above)
Comment (stuartwdouglas) [reply websocket/core/src/main/java/io/undertow/websockets/handshake/Base64.java:26]: Because this code was written to work on JDK7 for EAP6 :-)

**Pull request merged**